test/stamps.spec.ts

```diff
@@ -241,3 +241,16 @@ describe('topUpStamp', () => {
     expect(Number(stampExtended.amount)).toBeGreaterThan(Number(defaultAmount))
   })
 })
+
+describe('extendsCapacity', () => {
+  it('should extend stamps capacity', async () => {
+    const beeDebug = new BeeDebug(url)
+
+    const stamp = await buyNewStamp(defaultDepth, defaultAmount, beeDebug)
+
+    await topUpStamp(beeDebug, stamp.batchId, (Number(stamp.stamp.amount) * 2).toString())
```

Comment:
Hmmm, maybe @agazso or @AuHau can confirm it, but IIRC `topup` is additive instead of specifying the target `amount`. So the `* 2` is unnecessary here.

Reply:
Also as a general comment (maybe concerns @agazso the most), we would need an atomic topup + dilute operation on blockchain level in order to make this feature solid, otherwise one of these operations can fail (http network issue, gas issue, non-graceful shutdown, running out of funds) making stamps grow asymmetrically.

Reply:
Oops, meant this comments for `src/stamps.ts`, not the test 🙂

Reply:
I'll wait for @AuHau @agazso 's insight on this

Reply:
@Cafe137 I already test the additive property of topup call, and yes, no need to double up the amount

Reply:
Oh, good catch @Cafe137 about the additive side!

And yes, agree about the atomicity problem here, I am afraid though that with the current Bee API this is not possible for us to handle :-( https://github.com/ethersphere/bee/issues/3359 would help us with this.